src/drivers/intel/dmic.c

```diff
@@ -1401,11 +1401,11 @@ static void dmic_start(struct dai *dai)
 				CIC_CONTROL_SOFT_RESET_BIT, 0);
 	}
 
+	/* Set bit dai->index */
 	if (dmic->state == COMP_STATE_PREPARE)
-		(*uncached_dmic_active_fifos)++;
+		*uncached_dmic_active_fifos_mask |= BIT(dai->index);
```

Comment:
In theory you could just always set the bit here. No matter whether it's unpause or regular start, the bit should always be set. Hmm, but maybe it's better to avoid the uncached read/write here. With current code there is no risk of bugs as dmic->state is set explicitly in the code that is calling dmic_start().

Reply:
Yep, I was thinking to add a clean-up commit to simplify this.